parser.hh

```diff
@@ -388,6 +404,9 @@ struct ParserPosition
 	///iterator into the input.
 	Input::iterator it;
 
+	///user-meaningful filename.
+	std::string filename;
```

Comment:
This turns out to be a bit of faff... I'll look at it now that a meeting has ended.